### ISSUE TITLE
[BUG] `npm update` hangs for some dependencies for `npm>=10.4.0`

### ISSUE THREAD

**Issue by charlesbjohnson** [npm/cli#7491]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

After updating NPM from `10.2.4` to `10.5.2`, I noticed that `npm update` was hanging when updating a specific dependency in one of my projects. I was able to reproduce the issue in a reduced environment here:
- https://github.com/charlesbjohnson/npm-cli-v10-bug

Please see the README there for more details.

I've confirmed the issue is present on:
- Linux
- macOS
- Windows

And that the issue appears between `npm@10.3.0` and `npm@10.4.0`.

### Expected Behavior

`npm update` should not hang.

### Steps To Reproduce

Please see the steps to reproduce in this repository:
- https://github.com/charlesbjohnson/npm-cli-v10-bug

TL;DR:
- With `npm` >= `10.4.0`...
- Clone the repository...
- Run `npm install`
- Run `npm update follow-redirects`
- Observe that `npm update` hangs

### Environment

- npm: 10.7.0
- Node.js: 22.1.0
- OS Name: Amazon Linux 2023 (also macOS 14.4.1 and Windows 10)
- System Model Name: N/A
- npm config:
```ini
; node bin location = /home/ec2-user/.asdf/installs/nodejs/22.1.0/bin/node
; node version = v22.1.0
; npm local prefix = /home/ec2-user/npm-cli-v10-bug
; npm version = 10.7.0
; cwd = /home/ec2-user/npm-cli-v10-bug
; HOME = /home/ec2-user
```

**Comment by Tofandel** [comment]
This issue is still happening it seems

Related
#4028

**Comment by xmedeko** [comment]
Yep, downgrade to 10.3.0 has solved hang on Ubuntu. (On Windows, it's OK). 

I have a larger Electron.js project. `npm ci` runs fine, but when I try to install some package `npm i ...` in that project npm hangs. (When I try to install the same package in an empty directory, then it's OK.) Downgrade to npm 10.3.0 solved the issue.
